diff --git a/node_modules/@modelcontextprotocol/sdk/dist/cjs/shared/uriTemplate.js b/node_modules/@modelcontextprotocol/sdk/dist/cjs/shared/uriTemplate.js
index 8f8c74a..5576909 100644
--- a/node_modules/@modelcontextprotocol/sdk/dist/cjs/shared/uriTemplate.js
+++ b/node_modules/@modelcontextprotocol/sdk/dist/cjs/shared/uriTemplate.js
@@ -172,9 +172,8 @@ class UriTemplate {
         if (part.operator === '?' || part.operator === '&') {
             for (let i = 0; i < part.names.length; i++) {
                 const name = part.names[i];
-                const prefix = i === 0 ? '\\' + part.operator : '&';
                 patterns.push({
-                    pattern: prefix + this.escapeRegExp(name) + '=([^&]+)',
+                    pattern: `(?:[?&]`+this.escapeRegExp(name)+`=([^&]+))?`,
                     name
                 });
             }
@@ -184,7 +183,7 @@ class UriTemplate {
         const name = part.name;
         switch (part.operator) {
             case '':
-                pattern = part.exploded ? '([^/,]+(?:,[^/,]+)*)' : '([^/,]+)';
+                pattern = part.exploded ? '([^/,?&]+(?:,[^/,?&]+)*)' : '([^/?,&]+)';
                 break;
             case '+':
             case '#':
diff --git a/node_modules/@modelcontextprotocol/sdk/dist/esm/shared/uriTemplate.js b/node_modules/@modelcontextprotocol/sdk/dist/esm/shared/uriTemplate.js
index 8247f63..ece3054 100644
--- a/node_modules/@modelcontextprotocol/sdk/dist/esm/shared/uriTemplate.js
+++ b/node_modules/@modelcontextprotocol/sdk/dist/esm/shared/uriTemplate.js
@@ -169,9 +169,8 @@ export class UriTemplate {
         if (part.operator === '?' || part.operator === '&') {
             for (let i = 0; i < part.names.length; i++) {
                 const name = part.names[i];
-                const prefix = i === 0 ? '\\' + part.operator : '&';
                 patterns.push({
-                    pattern: prefix + this.escapeRegExp(name) + '=([^&]+)',
+                    pattern: `(?:[?&]`+this.escapeRegExp(name)+`=([^&]+))?`,
                     name
                 });
             }
@@ -181,7 +180,7 @@ export class UriTemplate {
         const name = part.name;
         switch (part.operator) {
             case '':
-                pattern = part.exploded ? '([^/,]+(?:,[^/,]+)*)' : '([^/,]+)';
+                pattern = part.exploded ? '([^/,?&]+(?:,[^/,?&]+)*)' : '([^/?,&]+)';
                 break;
             case '+':
             case '#':
